lib/elixir/lib/range.ex

```diff
@@ -40,7 +40,8 @@ defmodule Range do
   defstruct first: nil, last: nil
 
   @type t :: %Range{}
-  @type t(first, last) :: %Range{first: first, last: last}
+  @type t(first :: integer, last :: integer) :: %Range{first: integer, last: integer}
```

Comment:
Actually, after some research I found that `@type t(first :: integer, last :: integer) :: %Range{first: integer, last: integer}` should confusingly enough be valid.